src/rgw/rgw_data_sync.cc

```diff
@@ -3030,15 +3009,14 @@ class InitBucketShardStatusCR : public RGWCoroutine {
  public:
   InitBucketShardStatusCR(RGWDataSyncCtx* sc,
                          const rgw_bucket_sync_pair_info& pair,
-                         rgw_bucket_index_marker_info& info,
                          const BucketIndexShardsManager& marker_mgr)
-    : RGWCoroutine(sc->cct), sc(sc), pair(pair), info(info), marker_mgr(marker_mgr)
+    : RGWCoroutine(sc->cct), sc(sc), pair(pair), marker_mgr(marker_mgr)
   {}
   int operate(const DoutPrefixProvider *dpp) {
     reenter(this) {
       // try exclusive create with empty status
       objv.generate_new_write_ver(cct);
-      yield call(new InitCR(sc, pair, status, info, marker_mgr, objv, exclusive));
+      yield call(new InitCR(sc, pair, status, marker_mgr, objv, exclusive));
```

Comment:
as discussed in irc, i think it's safe to do away with the retry loop here in `InitBucketShardStatusCR`, and just do this initial write with `exclusive=false` so it never fails with `EEXIST` or `ECANCELED`

there may be racing writers, but Init should always win these races - this is accomplished by `objv.generate_write_new_ver()`, which will change the 'tag' in cls_version and cause other writers to fail on `cls_version_check()`

Reply:
ok. will do that in a separate commit, as this should not change functionality here